src/spec/test/metaprogramming/InterceptionThroughMetaClassTest.groovy

```diff
@@ -0,0 +1,23 @@
+package metaprogramming
+
+import groovy.xml.Entity
+
+// tag::meta_class_interception[]
+class InterceptionThroughMetaClassTest extends GroovyTestCase {
+    void testPOJOMetaClassInterception() {
+        String invoking = 'ha'
+        invoking.metaClass.invokeMethod = {String name, Object args ->
+            'invoked'
+        }
+        assertEquals invoking.length(), invoking.someMethod()
+    }
+
+    void testPOGOMetaClassInterception() {
+        Entity entity = new Entity('Hello')
+        entity.metaClass.invokeMethod = {String name, Object args ->
+            'invoked'
+        }
+        assertEquals entity.build(new Object()), entity.someMethod()
```

Comment:
I would also change this to `assert invoking.someMethod() == 'invoked'` instead. Imho this makes the intent a bit clearer. What do you think?